### PR TITLE
Add in support for more field tags, like maximum

### PR DESCRIPTION
**Pull request by wsong**

### CODE REVIEW

**Comment by rmohr** [comment]
I found it also very useful to reuse the [govalidator](https://github.com/asaskevich/govalidator/) tags, like it was done in the [abandoned swagger2 implementation](https://github.com/emicklei/swagger2/pull/2/commits/e2a1905dd0e0ceed7c26e783e48d87f738ac2822)

@emicklei @wsong what do you think?

**Comment by wsong** [comment]
Not a bad idea, but let's put that in a separate PR.

File: property_ext.go [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+}
+
+func setType(prop *spec.Schema, field reflect.StructField) {
+	if tag := field.Tag.Get("type"); tag != "" {
+		// Check if the first two characters of the type tag are
+		// intended to emulate slice/array behaviour.
+		//
+		// If type is intended to be a slice/array then add the
+		// overriden type to the array item instead of the main property
+		if len(tag) > 2 && tag[0:2] == "[]" {
```
Comment (emicklei) [inline property_ext.go]: i wonder if this magic is still needed with openapi spewc

File: property_ext_test.go [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+func TestThatExtraTagsAreReadIntoModel(t *testing.T) {
+	type fakeint int
+	type fakearray string
+	type Anything struct {
+		Name      string    `description:"name" modelDescription:"a test"`
+		Size      int       `minimum:"0" maximum:"10"`
+		Stati     string    `enum:"off|on" default:"on" modelDescription:"more description"`
+		ID        string    `unique:"true"`
+		FakeInt   fakeint   `type:"integer"`
+		FakeArray fakearray `type:"[]string"`
```
Comment (emicklei) [inline property_ext_test.go]: is the ```[]strings``` notation part of openapi or just a convenience for Go?  what is the JSON schema notation?  (will have a lookup later)

Comment (wsong) [inline property_ext_test.go]: To be honest, I assumed that you had added this `[]string` syntax in `go-restful-swagger12` yourself. I can take it out if you prefer.

Comment (emicklei) [inline property_ext_test.go]: that's true.  for this package we can re-evaluate that decision.

**Comment by wsong** [comment]
Any more thoughts on this PR?

**Pull request merged**